network/peer.go

```diff
@@ -836,3 +868,50 @@ func (p *peer) getIP() utils.IPDesc {
 	defer p.ipLock.RUnlock()
 	return p.ip
 }
+
+// assumes stateLock is held
+func (p *peer) addAlias(ip utils.IPDesc) {
+	p.ipLock.Lock()
+	defer p.ipLock.Unlock()
+
+	p.net.peerAliasIPs[ip.String()] = struct{}{}
+	p.aliases = append(p.aliases, alias{
+		ip:    ip,
+		added: p.net.clock.Time().Unix(),
+	})
+}
+
+// assumes stateLock is held
+func (p *peer) releaseAliases() {
+	p.ipLock.Lock()
+	defer p.ipLock.Unlock()
+
+	for _, alias := range p.aliases {
+		str := alias.ip.String()
+		delete(p.net.peerAliasIPs, str)
+
+		p.net.log.Verbo("released alias %s for peer %s", str, p.id.String())
```

Comment:
nit: you can omit `.String()` from `p.id.String()`

Reply:
Ahh, didn't realize `ids.ShortID` had a stringer